### PR TITLE
feat: partial automation of adding new release to other repos

### PR DESCRIPTION
**Pull request by wojciechka**
Adds partial automation for releasing updated version of telegraf-operator to streamline the process.

This creates a new branch, modifies the required files, and makes a commit. We may consider adding git push and creating a PR later.

### CODE REVIEW

File: scripts/release/update-community-operators.bash [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+cd "$DIR/../.."
+
+COMMUNITY_OPERATORS_PATH="$(pwd)/../community-operators"
+OPERATOR_PATH="operators/telegraf-operator"
+
+main() {
+  local version="${1:?Version required}"
+  local branch="${2:-telegraf-operator-v$(echo "${version}" | sed 's#\.#-#g')}"
+  local dir="${OPERATOR_PATH}/${version}"
+  local hygendir="$(pwd)"
```
Comment (mkmik) [inline scripts/release/update-community-operators.bash:15]:
```suggestion
  local hygendir="${PWD}"
```

File: scripts/release/update-community-operators.bash [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    exit 1
+  fi
+
+  cd "${COMMUNITY_OPERATORS_PATH}"
+  if [ "$(git rev-parse --abbrev-ref HEAD)" != "main" ] ; then
+    echo "community-operators repo at $(pwd) must be checked out as \"main\""
+    exit 1
+  fi
+
+  if [ ! -z "$(git status --porcelain)" ]; then
```
Comment (mkmik) [inline scripts/release/update-community-operators.bash:28]:
```suggestion
  if [ -n "$(git status --porcelain)" ]; then
```

File: scripts/release/update-community-operators.bash [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  local hygendir="$(pwd)"
+
+  if [ ! -d "${COMMUNITY_OPERATORS_PATH}/${OPERATOR_PATH}" ] ; then
+    echo >&2 "Unable to find ${COMMUNITY_OPERATORS_PATH}/${OPERATOR_PATH}; exiting..."
+    exit 1
+  fi
+
+  cd "${COMMUNITY_OPERATORS_PATH}"
+  if [ "$(git rev-parse --abbrev-ref HEAD)" != "main" ] ; then
+    echo "community-operators repo at $(pwd) must be checked out as \"main\""
```
Comment (mkmik) [inline scripts/release/update-community-operators.bash:24]: why here no `>&2` ?

File: scripts/release/update-helm-charts.bash [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  git pull
+  git checkout -b "${branch}"
+
+  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
+    -i "${CHARTS_FILE}"
+
+  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
+    -i "${CHARTS_FILE}"
+
```
Comment (mkmik) [inline scripts/release/update-helm-charts.bash:39]: so it works also on BSD sed (e.g. on macos)

```suggestion
  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
    -i '' "${CHARTS_FILE}"

  sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" \
    -i '' "${CHARTS_FILE}"

```

File: scripts/release/update.bash [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+cd "$DIR/../.."
+
+HELM_CHARTS_PATH="$(pwd)/helm-charts"
```
Comment (mkmik) [inline scripts/release/update.bash:8]:
```suggestion
HELM_CHARTS_PATH="${PWD}/helm-charts"
```

File: scripts/release/update-helm-charts.bash (hunk 4 above)
Comment (wojciechka) [reply scripts/release/update-helm-charts.bash:39]: Just out of curiosity - why is the `''` argument needed on macos?

Comment (wojciechka) [reply scripts/release/update-helm-charts.bash:39]: Ah I see, `-i` is something else on macos than Linux.

It seems there's no (easy?) way to call sed with in-place replacement on macos and linux.

This is why we can't just have nice things - anything beyond basic functionality is just different between OSes.

Comment (wojciechka) [reply scripts/release/update-helm-charts.bash]: Ok, changed it so it is not using the in-place editing of files, but a temporary file.

I am not happy with that, but it should work on both OSes.

Comment (mkmik) [reply scripts/release/update-helm-charts.bash]: the question is why is this optional on linux:

linux man page:

```
    -i[SUFFIX], --in-place[=SUFFIX]

              edit files in place (makes backup if SUFFIX supplied)
```

bsd man page:

```
     -i extension
             Edit files in-place, saving backups with the specified extension.  If a zero-length extension is given, no backup will be saved.  It is not recommended to give a zero-length extension
             when in-place editing files, as you risk corruption or partial content in situations where disk space is exhausted, etc.
```

File: scripts/release/update-helm-charts.bash [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  git checkout -b "${branch}"
+
+  # use a temporary file for updating version to
+  # ensure consistency between Linux and macOS
+  cat <"${CHARTS_FILE}" >"${CHARTS_FILE}.tmp"
+  cat <"${CHARTS_FILE}.tmp" | \
+    sed -E "s/^([[:space:]]*version:[[:space:]]*)[0-9].*\$/\1${version}/" | \
+    sed -E "s/^([[:space:]]*appVersion:[[:space:]]*v)[0-9].*\$/\1${version}/" \
+    >"${CHARTS_FILE}"
+  rm -f "${CHARTS_FILE}.tmp"
```
Comment (mkmik) [inline scripts/release/update-helm-charts.bash:41]: I think `-i''` works just fine in both macos and linux (no space between `i` and `''`)

Comment (mkmik) [reply scripts/release/update-helm-charts.bash:41]: hm no:

```
-i=''
```

File: scripts/release/update-helm-charts.bash (hunk 4 above)
Comment (wojciechka) [reply scripts/release/update-helm-charts.bash]: It is worse, on Linux it is optional, but if you want to provide value, it has to be non-empty and same argument.

So there is no way to invoke sed with in-place in a consistent. The only answer I found in https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux is to do `-i.bak` and then have a leftover `.bak` file or having to remove it.

Maybe that's better than my current approach?

Comment (mkmik) [reply scripts/release/update-helm-charts.bash]: (I don't like keeping tmp .bkp files unless we also clean up with a trap)

**Review by mkmik — APPROVED** [review]
I'm ok with this;

I still think that using a `-i=''` will be better (also because it helps our future selves have an example to look for when thinking about portability of sed)

**Comment by wojciechka** [comment]
> I still think that using a `-i=''` will be better (also because it helps our future selves have an example to look for when thinking about portability of sed)

I thought there is no way to get consistent `-i=''` behavior across macOS and Linux.

```
macos% echo a >/tmp/xyz ; sed "s/a/b/" -i='' /tmp/xyz ; uname -r ; cat /tmp/xyz
sed: -i=: No such file or directory
b
19.6.0
a
```

**Pull request merged**